tests/PySys/software-management-end-to-end/environment_sm_management.py

```diff
@@ -278,3 +334,37 @@ def check_is_installed(self, package_name, version=None):
                 ret = True
                 break
         return ret
+
+    def getpkgversion(self, pkg):
+        """ "Use apt-cache madison to derive a package version from
+        the apt cache even when it is not installed.
+        Not very bulletproof yet!!!
+        """
+        output = subprocess.check_output(["/usr/bin/apt-cache", "madison", pkg])
+
+        # Lets assume it is the package in the first line of the output
+        return output.split()[2].decode("ascii")  # E.g. "1.16-1+b3"
+
+    def get_pkgid(self, pkg):
+
+        # Database with package IDs
+        # TODO use this everywhere
+        pkgiddb = {
+            # apt
+            "asciijump": "5475278",
+            "robotfindskitten": "5473003",
+            "squirrel3": "5474871",
+            "rolldice": "5445239",
+            "moon-buggy": "5439204",
```

Comment:
Could we add something like read those from env variables?

Reply:
I added a proposal: We can now read the database as json from env var C8YSWREPO. There is a template command in the head of the file. Please check if it is convenient for you. In case of parser errors the user will see a exception traceback with info. We are still planning to read the IDs from C8y some day :)